plugin/advisor/catalog/walk_through.go

```diff
@@ -0,0 +1,420 @@
+package catalog
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/bytebase/bytebase/plugin/advisor/db"
+	"github.com/bytebase/bytebase/plugin/parser"
+
+	tidbparser "github.com/pingcap/tidb/parser"
+	tidbast "github.com/pingcap/tidb/parser/ast"
+	"github.com/pingcap/tidb/parser/format"
+	"github.com/pingcap/tidb/parser/model"
+)
+
+// WalkThroughErrorType is the type of WalkThroughError.
+type WalkThroughErrorType int
+
+const (
+	// PrimaryKeyName is the string for PK.
+	PrimaryKeyName string = "PRIMARY"
+	// FullTextName is the string for FULLTEXT.
+	FullTextName string = "FULLTEXT"
+
+	// 101 parse error type.
+
+	// ErrorTypeParseError is the error in parsing.
+	ErrorTypeParseError WalkThroughErrorType = 101
+	// ErrorTypeRestoreError is the error in restoring.
+	ErrorTypeRestoreError WalkThroughErrorType = 102
+
+	// 201 ~ 299 database error type.
+
+	// ErrorTypeAccessOtherDatabase is the error that try to access other database.
+	ErrorTypeAccessOtherDatabase = 201
+
+	// 301 ~ 399 table error type.
+
+	// ErrorTypeTableExists is the error that table exists.
+	ErrorTypeTableExists = 301
+
+	// 401 ~ 499 column error type.
+
+	// ErrorTypeColumnExists is the error that column exists.
+	ErrorTypeColumnExists = 401
+	// ErrorTypeColumnNotExists is the error that column not exists.
+	ErrorTypeColumnNotExists = 504
+
+	// 501 ~ 599 index error type.
+
+	// ErrorTypePrimaryKeyExists is the error that PK exists.
+	ErrorTypePrimaryKeyExists = 501
+	// ErrorTypeIndexExists is the error that index exists.
+	ErrorTypeIndexExists = 502
+	// ErrorTypeIndexEmptyKeys is the error that index has empty keys.
+	ErrorTypeIndexEmptyKeys = 503
+)
+
+// WalkThroughError is the error for walking-through.
+type WalkThroughError struct {
+	Type    WalkThroughErrorType
+	Content string
+}
+
+// NewParseError returns a new ErrorTypeParseError.
+func NewParseError(content string) *WalkThroughError {
+	return &WalkThroughError{
+		Type:    ErrorTypeParseError,
+		Content: content,
+	}
+}
+
+// Error implements the error interface.
+func (e *WalkThroughError) Error() string {
+	return e.Content
+}
+
+// WalkThrough will change the catalog schema by statements.
+func (d *databaseState) WalkThrough(stmts string) error {
+	if d.dbType != db.MySQL {
+		return nil
+	}
+
+	// For MySQL, there is only one schema whose name is empty string.
```

Comment:
Which special schema can have empty string?

Reply:
We define the `Catalog` as `Database -> Schema -> Table`. The `Schema` is only for PostgreSQL. So we use a `Schema` whose name is empty for other engines, such as MySQL.

Reply:
I think this explanation is more readable than the current comment. Please update it accordingly.